locations/Locations.py

```diff
@@ -0,0 +1,433 @@
+                'latitude': search.group(2)
+            }
+
+    def get_primary_id(self):
+        return self.guid
+
+    def _set_attributes(self):
+        attributes = {
+            'name': self.group_name,
+            'geoLocation': self.geo_location,
+            'address': self.street_address,
+            'city': self.city,
+            'state': self.state,
+            'zip': self.zip,
+            'telephone': self.telephone,
+            'fax': self.fax,
+            'county': self.county,
+            'website': self.location_url,
+            'type': self.type,
+            'campus': self.campus
+        }
+
+        for key, value in attributes.items():
+            self.update_attributes(key, value)
+
+
+class FacilLocation(Location):
+    """
+    The location type for facil locations
+    """
+    def __init__(self, raw_facil, raw_gir, raw_geo, proj):
+        """
+        Merge Banner locations with the data of gender inclusive restrooms and
+        geometries from ArcGIS
+        """
+        address1 = raw_facil.get('address1')
+        address2 = raw_facil.get('address2')
+
+        lon_lat = None
+        if raw_geo:
+            lon_lat = proj(
+                raw_geo['longitude'],
+                raw_geo['latitude'],
+                inverse=True
+            )
+
+        self._init_attributes()
+        self.bldg_id = raw_facil['id']
+        self.type = 'building'
+        self.tags = []
+        self.abbreviation = raw_facil.get('abbreviation')
+        self.name = raw_facil.get('name')
+        self.campus = self._get_pretty_campus(raw_facil.get('campus'))
+        self.address = self._get_address(address1, address2)
+        self.city = raw_facil.get('city')
+        self.state = raw_facil.get('state')
+        self.zip = raw_facil.get('zip')
+        self.geo_location = self._create_geo_location(
+            lon_lat[0] if lon_lat else None,
+            lon_lat[1] if lon_lat else None
+        )
+        self.geometry = self._create_geometry(
+            raw_geo['coordinatesType'] if raw_geo else None,
+            raw_geo['coordinates'] if raw_geo else None
+        )
+        self.gir_count = raw_gir['count'] if raw_gir else 0
+        self.gir_limit = bool(raw_gir['limit'].strip()) if raw_gir else None
+        self.gir_locations = raw_gir['all'].strip() if raw_gir else None
+        self.arcGIS_abbreviation = (
+            (raw_geo.get('abbreviation') if raw_geo else None)
+            or (raw_gir.get('abbreviation') if raw_gir else None)
+        )
+        self.relationships = {'services': {'data': []}}
+        self.merge = False
+        self.open_hours = {}
+        self.description = None
+        self.descriptionHtml = None
+        self.images = None
+        self.thumbnails = []
+        self.website = None
+        self.synonyms = None
+
+    def _get_pretty_campus(self, raw_campus):
+        campus_dict = {
+            'cascadescampus': 'Cascades',
+            'osucampus': 'Corvallis',
```

Comment:
I think this should be `osucorvallis` https://github.com/osu-mist/locations-api/blob/6e90fbd24d60aaf8d351cee1d4dfbb1e2e1452de/src/main/groovy/edu/oregonstate/mist/locations/core/FacilLocation.groovy#L62-L63

Reply:
Nice catch, fixed.